OpenRA.Game/Widgets/Widget.cs

```diff
@@ -237,6 +241,15 @@ public virtual void Initialize(WidgetArgs args)
 								   height);
 		}
 
+		protected void OnResolutionChange()
+		{
+			if (widgetArgs != null)
+				Initialize(widgetArgs);
```

Comment:
Widgets do more than just lay out their geometry in Inititialize.  Calling this multiple times is a bad idea.  You should instead implement a new `OnResolutionChangeInner` method that each widget can implement only the things it needs on resize (most widgets will be able to use a default implementation on the base widget).